### ISSUE TITLE
Proxy SwitchyOmega extension for Microsoft Edge (Chromium) - Official or unofficial?

### ISSUE THREAD

**Issue by maxkorsov** [FelisCatus/SwitchyOmega#2004]
At Microsoft Store, the Proxy SwitchyOmega extension for Microsoft Edge has appeared:
https://microsoftedge.microsoft.com/addons/detail/dijmmgblneagkcdganednabkbgjmceoe

There is no information that this is an official extension.
There are no links to Felis Catus developer and Proxy SwitchyOmega project page.

Can I trust this extension?

**Comment by KexyBiscuit** [comment]
FYI, the publisher of Proxy SwitchyOmega on Microsoft Edge Addons, `edge ext`, also [publishes a few extensions](https://microsoftedge.microsoft.com/addons/search/edge%20ext?hl=en-US) which doesn't belong to @FelisCatus. `edge ext` doesn't fill Privacy Policy URIs correctly either.

**Comment by maxkorsov** [comment]
I am very sorry that [@FelisCatus](https://github.com/FelisCatus) stopped working on his projects.
I can't trust - **edge ext**

**Comment by FelisCatus** [comment]
The Proxy SwitchyOmega [listing](https://microsoftedge.microsoft.com/addons/detail/dijmmgblneagkcdganednabkbgjmceoe) on Microsoft Edge Add-ons website is NOT uploaded by me, or anyone of the SwitchyOmega authors.

In other words, you may say the listing is **unofficial**. I have not tested or audited the uploaded extension package of that listing and I have zero idea whether the extension functions properly or whether it behaves responsibly.

It then follows common security practice to assume the worst for untrusted and non-audited code. If that's not enough, here goes ALL CAPS: USE CAUTION AND INSTALL AT YOUR OWN RISK!

**Comment by choyri** [comment]
TLDR: **DO NOT USE IT** ❌

- [Analysis of it by Chinese friends](https://www.v2ex.com/t/696314)
- [The content of the malicious script](https://gist.github.com/jat001/b10d12eee207c380246e7be74f8e6ed6)

**Comment by iamsurfing** [comment]
那你们就不能先考虑改一个名字再上架吗？
uBlock origin就是一个例子。

**Comment by MarkDDi** [comment]
官方已失效了，请作者尽快上传。谢谢。 未找到“switchyomega”的结果

**Comment by Artoria2e5** [comment]
Just use the Chrome store link at https://chrome.google.com/webstore/detail/proxy-switchyomega/padekgcemlokbadohgkifijomclgjgif. Microsoft Edge supports installing from Chrome store.

**Comment by BluntMyLife** [comment]
I've used the malicious extension from the edge store probably for 6+ months. Can someone tell me please what the malicious code done to me? What should I do know after uninstalling the extension?

**Comment by huanghe2015** [comment]
How about the plan on uploading to Edge Addon Store?

**Comment by huanghe2015** [comment]
If the author doesn't want to solve this, I will have a try.

**Comment by huanghe2015** [comment]
Done. https://microsoftedge.microsoft.com/addons/detail/proxy-switchyomega/fdbloeknjpnloaggplaobopplkdhnikc

**Comment by huanghe2015** [comment]
Anyone can review the library I changed to ensure whether it has malicious code. If not, the extension I had uploaded is safe.

**Comment by Wirili** [comment]
> Anyone can review the library I changed to ensure whether it has malicious code. If not, the extension I had uploaded is safe.
> https://github.com/powwowinc/angular-spectrum-colorpicker

色块选择叠在一起了，能修复下吗

**Comment by huanghe2015** [comment]
> 
> 
> > Anyone can review the library I changed to ensure whether it has malicious code. If not, the extension I had uploaded is safe.
> > https://github.com/powwowinc/angular-spectrum-colorpicker
> 
> 色块选择叠在一起了，能修复下吗

等我有空吧。现在刷题忙。

**Comment by huanghe2015** [comment]
> 
> 
> Just use the Chrome store link at https://chrome.google.com/webstore/detail/proxy-switchyomega/padekgcemlokbadohgkifijomclgjgif. Microsoft Edge supports installing from the Chrome store.

The extension you mentioned has an extra JS file that will hijack your link on some e-commerce websites to his affiliate link to earn money.

**Comment by huanghe2015** [comment]
> 
> 
> I've used the malicious extension from the edge store probably for 6+ months. Can someone tell me please what the malicious code done to me? What should I do now after uninstalling the extension?

The extension you mentioned has an extra JS file that will hijack your link on some e-commerce websites to his affiliate link to earn money.

**Comment by huanghe2015** [comment]
A retry build based on the master branch will be posted after being reviewed.

**Comment by huanghe2015** [comment]
Sorry for the delay, tried to upload the code on the master branch is rejected, still trying.

**Comment by yobai** [comment]
> 
这个是大佬自己上传的吗？是可以安全使用的吧

**Comment by yobai** [comment]
> Done. https://microsoftedge.microsoft.com/addons/detail/proxy-switchyomega/fdbloeknjpnloaggplaobopplkdhnikc

请问，现在edge扩展商店的Proxy SwitchyOmega插件，是您上传的吗？是否可以安心使用呢？

**Comment by huanghe2015** [comment]
> > Done. https://microsoftedge.microsoft.com/addons/detail/proxy-switchyomega/fdbloeknjpnloaggplaobopplkdhnikc
> 
> 请问，现在edge扩展商店的Proxy SwitchyOmega插件，是您上传的吗？是否可以安心使用呢？

可以看这段

https://github.com/FelisCatus/SwitchyOmega/issues/2004#issuecomment-739817695

**Comment by yobai** [comment]
> > > Done. https://microsoftedge.microsoft.com/addons/detail/proxy-switchyomega/fdbloeknjpnloaggplaobopplkdhnikc
> > 
> > 
> > 请问，现在edge扩展商店的Proxy SwitchyOmega插件，是您上传的吗？是否可以安心使用呢？
> 
> 可以看这段
> 
> [#2004 (comment)](https://github.com/FelisCatus/SwitchyOmega/issues/2004#issuecomment-739817695)

好的，谢谢。那您就是CubeK本人吧，感谢您提供的插件

**Comment by huanghe2015** [comment]
> > > > Done. https://microsoftedge.microsoft.com/addons/detail/proxy-switchyomega/fdbloeknjpnloaggplaobopplkdhnikc
> > > 
> > > 
> > > 请问，现在edge扩展商店的Proxy SwitchyOmega插件，是您上传的吗？是否可以安心使用呢？
> > 
> > 
> > 可以看这段
> > [#2004 (comment)](https://github.com/FelisCatus/SwitchyOmega/issues/2004#issuecomment-739817695)
> 
> 好的，谢谢。那您就是CubeK本人吧，感谢您提供的插件

主要是现在转向[v2rayN](https://github.com/2dust/v2rayN)之后我就不怎么管这块了。我手里的免费SS和SSR都死翘了，去掉[PAC](https://github.com/2dust/v2rayN/issues/1366)之后这个插件在我手中基本没啥价值了。当然，以后如果要弄[家庭网关](https://github.com/twotreesus/V2RayPi)的话PAC还是有价值的。

**Comment by yobai** [comment]
> 

确实看您最近不太管这一块了，edge扩展商店下面评论说这个插件会在购物网站劫持跳转，话说得挺不友善的，我自己使用没有出现这个问题。感觉他们是不是把这个插件和之前套壳的插件混淆了，还以为是您上传的插件存在套壳。

**Comment by yobai** [comment]
> 主要是现在转向[v2rayN](https://github.com/2dust/v2rayN)之后我就不怎么管这块了。我手里的免费SS和SSR都死翘了，去掉[PAC](https://github.com/2dust/v2rayN/issues/1366)之后这个插件在我手中基本没啥价值了。

确实看您不太管这块了，edge扩展商店的评论并不友善，说是存在购物网站劫持，我自己使用没有啥问题，估计是他们将套壳的插件（https://microsoftedge.microsoft.com/addons/detail/dijmmgblneagkcdganednabkbgjmceoe）和这个插件弄混了。

**Comment by huanghe2015** [comment]
上传我的后台截图作为证明（有效期7天）
I have uploaded the screenshot of my admin console as the witness with the expiration of 7 days.
https://zapyatransfer.com/r/OHhq0i0auVVIECVYFwk

**Comment by KexyBiscuit** [comment]
> 上传我的后台截图作为证明（有效期7天） I have uploaded the screenshot of my admin console as the witness with the expiration of 7 days. https://zapyatransfer.com/r/OHhq0i0auVVIECVYFwk
Just a re-upload for backup.
![screenshot20230703](https://github.com/FelisCatus/SwitchyOmega/assets/8013154/43960e46-de41-45bd-9ae4-5582354cb773)
